c++/src/kj/compat/http.c++

```diff
@@ -6346,12 +6346,15 @@ private:
           streamAndFulfiller(initStreamsAndFulfiller(kj::mv(stream))) {}
 
     ~ConnectResponseImpl() noexcept(false) {
-      if (fulfiller->isWaiting()) {
-        fulfiller->reject(KJ_EXCEPTION(DISCONNECTED, "the connect request was incomplete"));
-      }
-      if (streamAndFulfiller.fulfiller->isWaiting()) {
-        streamAndFulfiller.fulfiller->reject(
-            KJ_EXCEPTION(DISCONNECTED, "the connect request was incomplete"));
+      if (fulfiller->isWaiting() || streamAndFulfiller.fulfiller->isWaiting()) {
+        auto ex = KJ_EXCEPTION(DISCONNECTED,
```

Comment:
This actually seems more like a FAILED than a DISCONNECTED to me. The server implementation is buggy if it didn't call the callback.

Reply:
Makes sense